docs/content/cookbook/handling-custom-occ-endpoint.md

```diff
@@ -0,0 +1,183 @@
+  customField: any;
+}
+
+export interface CustomResponse {
+  whatever: any;
+}
+
+const callCustomEndpoint = async (
+  context: SapccIntegrationContext,
+  props: CustomEndpointProps
+): Promise<CustomResponse> => {
+  const { config, req, client } = context;
+
+  const userId = getUserIdFromRequest({ req, props } as any); // retrieves userID from props or cookies
+
+  const res = await client.get(
+    `/${config.api.baseSiteId}/users/${userId}/customEndpoint/${props.customField}`,
+    createRequestOptions({
+      // adds authorization headers and language & currency parameters
+      context,
+      props,
+      tokenMode: TokenModes.CUSTOMERORAPPLICATION,
+    })
+  );
+
+  return res.data;
+};
+```
+
+Read more about the helper methods:
+
+- [getUserIdFromRequest](https://docs.alokai.com/integrations/sapcc/api/sapcc-api/getUserIdFromRequest)
+- [createRequestOptions](https://docs.alokai.com/integrations/sapcc/api/sapcc-api/createRequestOptions)
+
+## Real life example
+
+Here's an example implementation of the product interest feature. This feature is available in OCC API, but not in the middleware and SDK integration.
+Let's add support for it.
+
+First, you need to add a new API method in the middleware. (For simplicity, this guide shows how to do it in one file, but we recommend splitting it into multiple files to maintain cleaner code.)
+
+```typescript [storefront-middleware/middleware.config.ts]
+import {
+  SapccIntegrationContext,
+  TokenModes,
+  createRequestOptions,
+  getUserIdFromRequest,
+} from "@vsf-enterprise/sapcc-api";
+import { BaseProps, BaseUserId, Product } from "@vsf-enterprise/sapcc-types";
+
+export interface GetProductInterestsProps extends BaseProps, BaseUserId {
+  productCode?: string;
+}
+export interface ProductInterestEntry {
+  interestType?: string;
+  dateAdded?: string;
+  expirationDate?: string;
+}
+export interface ProductInterestRelation {
+  product?: Product;
+  productInterestEntry?: Array<ProductInterestEntry>;
+}
+export interface UserInterestsResponse {
+  results: Array<ProductInterestRelation>;
+}
+
+const getProductInterests = async (
+  context: SapccIntegrationContext,
+  props: GetProductInterestsProps
+): Promise<UserInterestsResponse> => {
+  const { config, req, client } = context;
+
+  const userId = getUserIdFromRequest({ req, props } as any);
+
+  const requestOptions = createRequestOptions({
+    context,
+    props,
+    tokenMode: TokenModes.CUSTOMERORAPPLICATION,
+  });
+
+  const res = await client.get(
+    `/${config.api.baseSiteId}/users/${userId}/productinterests`,
+    {
+      ...requestOptions,
+      params: {
+        ...requestOptions.params,
+        productCode: props.productCode,
+      },
+    }
+  );
+
+  return res.data;
+};
+
+const apiMethods = methods<typeof normalizers>();
+const unifiedApiExtension = createUnifiedExtension<Context, Config>()({
+  normalizers,
+  apiMethods: {
+    ...apiMethods,
+    getProductInterests,
```

Comment:
@mateuszo please add `getProductInterests` as a separate extension

https://github.com/vuestorefront/unified-storefronts/blob/next/1.0.0-docs/docs/content/2.integration-%26-setup/0.creating-new-api-methods.md#adding-new-api-methods

Reply:
I did it this way the first time but it messed up the typing a little bit. I will do it your way in the next iteration because I need more time to figure out how to type it correctly.